### PR TITLE
🧠 Forgot password flow

### PR DESCRIPTION
**Pull request by WRadoslaw**
#4190

### CODE REVIEW

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            loader
+            hasNavigatedBack
+          />
+        )}
+      </FormProvider>
+    </DialogModal>
+  )
+}
+
+// dwad!aA@13
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx]: 👽 ✋🏻

File: packages/atlas/src/components/_auth/ForgotPasswordModal/steps/NewPasswordStep.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          icon: isFieldVisible[name] ? <SvgActionHide /> : <SvgActionShow />,
+          onClick: () => setIsFieldVisible((fields) => ({ ...fields, [name]: !fields[name] })),
+        },
+      }
+    },
+    [isFieldVisible, register]
+  )
+
+  return (
+    <AuthenticationModalStepTemplate hasNavigatedBack={false} title="Change password" subtitle="NewPasswordStep">
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ForgotPasswordModal/steps/NewPasswordStep.tsx]:
```suggestion
    <AuthenticationModalStepTemplate hasNavigatedBack={false} title="Change password" subtitle="Set your new password.">
```

File: packages/atlas/src/components/_inputs/FormField/FormField.tsx [hunk 3]
```diff
@@ -43,6 +44,7 @@ export const FormField = memo(
         tooltip,
         error,
         disableErrorAnimation,
+        footerNode,
```
Comment (drillprop) [inline packages/atlas/src/components/_inputs/FormField/FormField.tsx]: To be honest I don't like the fact that we're adding another prop to the component just to add the `Forgot password?` link. I don't see any other situation where we will need this in the future. Why don't we just put the link below the FormField component?

File: packages/atlas/src/providers/auth/auth.helpers.ts [hunk 4]
```diff
@@ -99,9 +101,54 @@ export const decodeSessionEncodedSeedToMnemonic = async (encodedSeed: string) =>
...
   return hasAtleastOneWallet ? 'externalLogIn' : 'logIn'
 }
+
+type EncryptSeedParams = {
+  seed: string
+  email: string
+  password: string
+}
+
+export const encryptSeed = async ({ seed, email, password }: EncryptSeedParams) => {
```
Comment (drillprop) [inline packages/atlas/src/providers/auth/auth.helpers.ts]: The name is misleading. We will return from this function encryption artficats, not only seed. 
I actually created a very similar function here https://github.com/Joystream/atlas/pull/4326/files#diff-6dc99f6da6a82344e4dd005fd116a64ae23fd9022aaddefdb48bb42577f7942aR17
And I called it prepareEncryptionArtifacts(I took the name from the orion)

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+      } catch (error) {
+        displaySnackbar({
+          title: 'Something went wrong',
+          description: `We encountered unexpected error. Please try again.`,
+          iconType: 'error',
+        })
+        SentryLogger.error('Failed to change password', 'ForgotPasswordModal', error)
+      } finally {
+        setCurrentStep(ForgotPasswordStep.NewPasswordStep)
+      }
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx:173]: Correct me if I'm wrong, but we don't want to change the step when there is an error.

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+      const loginSignature = await keypair.sign(JSON.stringify(loginPayload))
+      const loginResponse = await loginRequest(u8aToHex(loginSignature), loginPayload)
+      const accData = await lazyCurrentAccountQuery()
+
+      if (accData.data?.accountData.email !== data['EmailAndSeedStep'].email) {
+        form.setError(`${ForgotPasswordStep.EmailAndSeedStep}.email`, {
+          message: 'Provided email do not match mnemonic.',
+        })
+        logoutRequest().catch(() => undefined)
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx]: Why `catch(() => undefined)`? Isn't it the same as not adding a catch at all?

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+      SentryLogger.error('Failed to login when recovering password', 'ForgotPasswordModal', error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleNewPasswordStep = useCallback(
+    async (data: ForgotPasswordModalForm) => {
+      if (!accountId) {
+        return
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx:122]: I would throw an error here, and wrap this whole code with try-catch, not only the axios part, there are a lot of things that can go wrong here.

File: packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts [hunk 8]
```diff
@@ -1,9 +1,14 @@
...
 import { sizes } from '@/styles'
 
 export const Container = styled.div`
   display: flex;
   flex-direction: column;
   gap: ${sizes(6)};
 `
+
+export const ForgotPasswordText = styled(TextButton)`
+  margin: 0 0 0 auto;
```
Comment (drillprop) [inline packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts]: `margin-left: auto`  would be much easier to read

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts (hunk 8 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts]: I needed to reset other margins to 0, that's the reason

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx (hunk 5 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx:173]: If we don't user will be stuck on laoding modal -> line 119

Comment (drillprop) [reply packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx:173]: ok, that make sense

File: packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts (hunk 8 above)
Comment (drillprop) [reply packages/atlas/src/components/_auth/LogInModal/LogInModal.styles.ts]: so you're saying that TextButton has some default margins applied?

File: packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx (hunk 6 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx]: Nope, this way it won't trigger catch which is there to handle login error

Comment (WRadoslaw) [reply packages/atlas/src/components/_auth/ForgotPasswordModal/ForgotPasswordModal.tsx]: I'll add sentry there as well for consistency

**Review by drillprop — APPROVED** [review]

**Pull request merged**